hal/network/ncp/at_parser/at_parser_impl.cpp

```diff
@@ -600,6 +600,7 @@ int AtParserImpl::nextLine(unsigned* timeout) {
 int AtParserImpl::readMore(unsigned* timeout) {
     assert(bufPos_ < INPUT_BUF_SIZE);
     const auto strm = conf_.stream();
+    CHECK_TRUE(strm, SYSTEM_ERROR_INVALID_STATE);
```

Comment:
I did hit it, because we de-initialized the modem serial if modem is to be powered off, in which case if user called `Cellular.command()` it will cause sos. Though I added `parser_.destroy()` when the modem is powered off, it won't hurt anything after adding this sanity check.

Reply:
I'm afraid I'll disagree. We are masking a bug in the `cellular_command()` implementation. It's not consulting with the NCP client state and that's the reason why it's able to access the parser with an invalid (no stream) state.

Reply:
So let's check the power state of modem in `cellular_command()` like this: `CHECK_TRUE(client->ncpPowerState() == NcpPowerState::ON, SYSTEM_ERROR_INVALID_STATE);` and not destroy the AT parser?

Reply:
It's not enough to check the power state. An AT command can be issued only if the client is an ON and ready state. See how other NCP client methods deal with this (e.g. `SaraNcpClient::getIccid`).